### ISSUE TITLE
individual column searching

### ISSUE THREAD

**Issue by kife-design** [yajra/laravel-datatables#562]
Hello, 

In issue #25 it is mentioned that it is possible to add single column searching and an example was added. Is this still valid, as i cant seem to find the demo with this example, nor am i able to implement this with the datatables. I keep getting the `this.api(...).columns(...).every is not a function` javascript error when implementing this into my code.

**Comment by yajra** [comment]
Yes it's possible, see this [demo](http://datatables.yajrabox.com/eloquent/multi-filter-select) for ref.

**Comment by kife-design** [comment]
But i am using the Datatables Service classes, and in this case it is not working.

**Comment by yajra** [comment]
You need to add a footer on your table, try passing a second param on table method of true like to draw the footer.

```php
{{ $dataTable->table(['class' => 'table'], true) }}
```

**Comment by yajra** [comment]
See this [PR](https://github.com/yajra/laravel-datatables/pull/471) for some ref.

**Comment by kife-design** [comment]
I dont think that this is the problem, i added the footer but i'm still getting this issue.

this is my getBuilderParameters as requested by the DataTables class:

```
protected function getBuilderParameters()
    {
        return [
            'order'   => [[1, 'desc']],
            'initComplete' => "function () {
                this.api().columns().every(function () {
                var column = this;
                var input = document.createElement(\"input\");
                $(input).appendTo($(column.footer()).empty())
                .on('change', function () {
                    column.search($(this).val(), false, false, true).draw();
                });
            });
            }"
        ];
    }
```

The error is throw on the first line of the initComplete function -> 'this.api().columns() is not a function'

Has someone already have a working example of column searching through the DataTables Service class?

**Comment by mtx-z** [comment]
Hey, i'm actually trying to implement this demo. 
Does we need any server processing ? I just make the same exact code with my model, and all search inputs make the table refresh with no data (but it should). 
Anything i miss ? thx

**Comment by yajra** [comment]
I just implement this issue on demo https://datatables.yajrabox.com/users and works well for me. Thanks!

**Comment by mtx-z** [comment]
Thanks a lot for answer. I finally did it :) 
But i'm using the [non-POST method](https://datatables.yajrabox.com/eloquent/multi-filter-select). 

I cannot get the [POST way](https://datatables.yajrabox.com/eloquent/post-column-search) to get working. It seems that the code in the `filterColumn` is never triggered (also tried with the same .`whereRaw` code but result is unchanged)

Previously, i was using the `->filter(function ($query)` to do something like : 
```
  $user_id = Input::get('id');
                 if ($user_id) {
                     $query->where('users.id', 'like', "%$user_id%");
                 }
```
And it is working perfectly. But doing this, i override and lost the default text search. 

I tried the same in `filterColumn` but nothing change, even a dd() dont do anything. 
So : 
- Am I doing something wrong with the `filterColumn` ?
- Is the [POST demo](https://datatables.yajrabox.com/eloquent/post-column-search) only have code for ID column filtering (in code sample) even if all column are searchable in demo ? (I only see ID related code in the filter part of the code sample provided).
- Latest solution: how can i in my override of `->filter` re-implement the global text search (in case i cannot do it with `filterColumn` ?

Thanks a lot, really !
